ui/apps/platform/src/hooks/useFetchReport.ts

```diff
@@ -1,31 +1,104 @@
 import { useEffect, useState } from 'react';
+import { fetchAccessScopes } from 'services/AccessScopesService';
+import { getCollection } from 'services/CollectionsService';
 
 import { fetchReportById } from 'services/ReportsService';
 import { ReportConfiguration } from 'types/report.proto';
+import useFeatureFlags from './useFeatureFlags';
 
-type Result = { isLoading: boolean; report: ReportConfiguration | null; error: string | null };
+export type ReportScope = {
+    // The 'AccessControlScope' type is deprecated and should be able to be removed in the release after 3.74
+    type: 'CollectionScope' | 'AccessControlScope';
+    id: string;
+    name: string;
+    description: string;
+};
 
-const defaultResultState = { report: null, error: null, isLoading: true };
+type Result = {
+    isLoading: boolean;
+    report: ReportConfiguration | null;
+    reportScope: ReportScope | null;
+    error: string | null;
+};
+
+const defaultResultState = {
+    report: null,
+    reportScope: null,
+    error: null,
+    isLoading: true,
+};
+
+function fetchCollectionReportScope(scopeId: string): Promise<ReportScope> {
+    const { request } = getCollection(scopeId);
+
+    return request.then(({ collection: { id, name, description } }) => ({
+        type: 'CollectionScope',
+        id,
+        name,
+        description,
+    }));
+}
+
+function fetchAccessScopeReportScope(scopeId: string): Promise<ReportScope | null> {
+    return fetchAccessScopes().then((scopes) => {
```

Comment:
Oh, good. Plural fetch and find prevents 404 status for a singular fetch!

Reply:
To be honest I was looking for a singular fetch and couldn't seem to find one for access scopes. I was going to let the promise reject normally and fall through to the error case.